src/main/java/net/snowflake/client/core/auth/wif/WorkloadIdentityUtil.java

```diff
@@ -1,6 +1,70 @@
 package net.snowflake.client.core.auth.wif;
 
+import com.nimbusds.jwt.JWT;
+import com.nimbusds.jwt.JWTParser;
+import java.util.HashMap;
+import java.util.Map;
+import net.snowflake.client.log.SFLogger;
+import net.snowflake.client.log.SFLoggerFactory;
+
 class WorkloadIdentityUtil {
+
+  private static final SFLogger logger = SFLoggerFactory.getLogger(WorkloadIdentityUtil.class);
+
   static final String SNOWFLAKE_AUDIENCE_HEADER_NAME = "X-Snowflake-Audience";
   static final String SNOWFLAKE_AUDIENCE = "snowflakecomputing.com";
+
+  static JwtClaims extractAndVerifyClaims(String token) {
```

Comment:
I'd strongly suggest against the term `verify` here, or even explicitly call it `extract_unverified_sub_and_iss`, since "verify" in this context almost always means check the signature, which we're not doing here. This is untrusted user input and callers of this function shouldn't assume otherwise.

For reference, I called the corresponding Python function [`extract_iss_and_sub_without_signature_verification`](https://github.com/snowflakedb/snowflake-connector-python/blob/5e621837ec1acbc0ddae1274c24ad314c0238b92/src/snowflake/connector/wif_util.py#L71).

```suggestion
  static JwtClaims extractUnverifiedSubAndIss(String token) {
```

Reply:
Actually true, I was basically checking for nulls only, "verification" would be a stretch. Renamed.

Reply:
Thanks! If you're open to it, it may still be good to explicitly include `unverified` in there somewhere to avoid misunderstandings, but it's not a big deal. Thank you for removing verified!

Reply:
Sure, renamed it to `extractClaimsWithoutVerifyingSignature`